corehq/apps/users/models.py

```diff
@@ -1346,7 +1346,7 @@ def get_by_username(cls, username, strict=True):
 
         def get(stale, raise_if_none):
             result = cls.get_db().view('users/by_username',
-                key=username,
+                key=username.lower(),
```

Comment:
I don't think that we want to do this. I think we want to keep authentication case sensitive. (my musing over slack might have been confusing)

Reply:
ah ok, how come you want to keep it case sensitive- is it because this is used in so many places?

Reply:
That seems like a big change that would need more thinking through. And i don't think that it's an unreasonable expectation to have it be case sensitive

Reply:
makes sense:
https://github.com/dimagi/commcare-hq/pull/20926/commits/b84597f5df4fca05e9f7a13a4f5a347a9a881d1e